service/src/main/kotlin/fi/tampere/trevaka/invoice/service/TrevakaInvoiceClient.kt

```diff
@@ -90,38 +91,30 @@ class TrevakaInvoiceClient(
     }
 
     private fun toHeader(invoice: InvoiceDetailed): Header {
+        val headOfFamily = toInvoicePerson(invoice.headOfFamily)
+        val codebtor = invoice.codebtor?.let { toInvoicePerson(it) }
+
         return Header().apply {
             customer = P1PartnerType().apply {
                 person = Person().apply {
-                    ssn = invoice.headOfFamily.ssn
+                    ssn = headOfFamily.ssn
                     personName = PersonName().apply {
-                        firstNames = invoice.headOfFamily.firstName
-                        surName = invoice.headOfFamily.lastName
-                    }
-                }
-                address = when (invoice.headOfFamily.restrictedDetailsEnabled) {
-                    true -> restrictedAddress
-                    false -> Address().apply {
-                        street = invoice.headOfFamily.streetAddress
-                        town = invoice.headOfFamily.postOffice
-                        postCode = invoice.headOfFamily.postalCode
+                        firstNames = headOfFamily.firstName
+                        surName = headOfFamily.lastName
                     }
                 }
+                address = headOfFamily.address()
             }
-            alternativePayer = when (hasAlternativePayer(invoice.headOfFamily)) {
+            alternativePayer = when (codebtor != null) {
```

Comment:
Has to be checked with integration partner if it is a must to always have the AlternativePayer element set even if there is not a codebtor. We'll change this later on, if it turns out so.